R/images.R

```diff
@@ -0,0 +1,204 @@
+#'   generated \code{<img>} (e.g. \code{id}, \code{class}).
+#' @param suppressSize By default, \code{plotTag} will include a \code{style}
+#'   attribute with \code{width} and \code{height} properties specified in
+#'   pixels. If you'd rather specify the image size using other methods (like
+#'   responsive CSS rules) you can use this argument to suppress width
+#'   (\code{"x"}), height (\code{"y"}), or both (\code{"xy"}) properties.
+#'
+#' @return A \code{\link{browsable}} HTML \code{<img>} tag object. Print it at
+#'   the console to preview, or call \code{\link{as.character}} on it to view the HTML
+#'   source.
+#'
+#' @seealso \code{\link{capturePlot}} saves plots as an image file.
+#'
+#' @examples
+#'
+#' img <- plotTag({
+#'   plot(cars)
+#' }, "A plot of the 'cars' dataset", width = 375, height = 275)
+#'
+#' if (interactive()) img
+#'
+#'
+#' svg <- plotTag(plot(pressure), "A plot of the 'pressure' dataset",
+#'   device = grDevices::svg, width = 375, height = 275, pixelratio = 1/72,
+#'   mimeType = "image/svg+xml")
+#'
+#' if (interactive()) svg
+#'
+#' @export
+plotTag <- function(expr, alt, device = defaultPngDevice(), width = 400, height = 400,
+  pixelratio = 2, mimeType = "image/png", deviceArgs = list(), attribs = list(),
+  suppressSize = c("none", "x", "y", "xy")) {
+
+  suppressSize <- match.arg(suppressSize)
+  if (suppressSize == "xy") {
+    suppressSize <- c("x", "y")
+  }
+
+  file <- rlang::eval_tidy(rlang::expr(capturePlot({{expr}},
+    device = device,
+    width = width * pixelratio,
+    height = height * pixelratio,
+    res = 72 * pixelratio,
+    !!!deviceArgs)))
+
+  on.exit(unlink(file), add = TRUE, after = FALSE)
+
+  browsable(tags$img(
+    src = base64enc::dataURI(file = file, mime = mimeType),
+    style = css(
+      width = if (!"x" %in% suppressSize) validateCssUnit(width),
+      height = if (!"y" %in% suppressSize) validateCssUnit(height)
+    ),
+    alt = alt,
+    !!!attribs
+  ))
+}
+
+
+#'   \code{\link[grDevices:png]{grDevices::png()}},
+#'   \code{\link[ragg:agg_png]{ragg::agg_png()}}, or
+#'   \code{\link[Cairo:CairoPNG]{Cairo::CairoPNG()}}, depending on your system
+
+#' Determine the best PNG device for your system
+#'
+#' Returns the best PNG-based graphics device for your system, in the opinion of
+#' the \code{htmltools} maintainers. On Mac,
+#' \code{\link[grDevices:png]{grDevices::png()}} is used; on all other
+#' platforms, either \code{\link[ragg:agg_png]{ragg::agg_png()}} or
+#' \code{\link[Cairo:CairoPNG]{Cairo::CairoPNG()}} are used if their packages
+#' are installed. Otherwise, \code{\link[grDevices:png]{grDevices::png()}} is
+#' used.
+#'
+#' @return A graphics device function.
+#'
+#' @export
+defaultPngDevice <- function() {
+  if (capabilities("aqua")) {
+    grDevices::png
+  } else if (nchar(system.file(package = "ragg"))) {
+    ragg::agg_png
+  } else if (nchar(system.file(package = "Cairo"))) {
```

Comment:
I think you want either `nchar(system.file(package = "Cairo")) > 0` or `system.file(package = "Cairo") != ""` (I prefer the latter)